src/hotspot/os/windows/os_windows.cpp

```diff
@@ -4482,8 +4482,18 @@ bool os::same_files(const char* file1, const char* file2) {
     return true;
   }
 
-  HANDLE handle1 = create_read_only_file_handle(file1);
-  HANDLE handle2 = create_read_only_file_handle(file2);
+  char* native_file1 = os::strdup(file1);
+  native_file1 = os::native_path(native_file1);
+  char* native_file2 = os::strdup(file2);
+  native_file2 = os::native_path(native_file2);
```

Comment:
`os::strdup` should `be os::strdup_check_oom`

Other than that, this PR looks good.

Reply:
Fixed. I'll do some testing before integrate.